traffic_ops/traffic_ops_golang/trafficvault/backends/postgres/postgres.go

```diff
@@ -156,6 +158,69 @@ func (p *Postgres) GetDeliveryServiceSSLKeys(xmlID string, version string, tx *s
 	return sslKey, true, nil
 }
 
+// GetExpirationInformation returns the expiration information for all SSL Keys.
+func (p *Postgres) GetExpirationInformation(tx *sql.Tx, ctx context.Context, days string) ([]tc.SSLKeyExpirationInformation, bool, error) {
+	tvTx, dbCtx, cancelFunc, err := p.beginTransaction(ctx)
+	if err != nil {
+		return []tc.SSLKeyExpirationInformation{}, false, err
+	}
+	defer p.commitTransaction(tvTx, dbCtx, cancelFunc)
+
+	var fedList []string
+	fedRows, err := tx.Query("SELECT DISTINCT(ds.xml_id) FROM federation_deliveryservice AS fd JOIN deliveryservice AS ds ON ds.id = fd.deliveryservice")
+	if err != nil {
+		return []tc.SSLKeyExpirationInformation{}, false, err
+	}
+	defer fedRows.Close()
+
+	for fedRows.Next() {
+		var fedString string
+		if err = fedRows.Scan(&fedString); err != nil {
+			return []tc.SSLKeyExpirationInformation{}, false, err
+		}
+		fedList = append(fedList, fedString)
+	}
+
+	query := "SELECT deliveryservice, cdn, provider, expiration FROM sslkey WHERE version='latest'"
+
+	if days != "" {
+		query = query + " AND expiration <= (now() + '" + days + " days'::interval)"
+	}
+
+	var expirationInfos []tc.SSLKeyExpirationInformation
```

Comment:
This is what I'm talking about right here; the "zero value" of a slice is `nil`, so if you don't find any expiration key info within the selected interval or where `version` is `'latest'`, then this function returns `nil` in its first return value.